rust/zypp-agama/src/lib.rs

```diff
@@ -95,15 +99,24 @@ pub fn list_repositories() -> Vec<Repository> {
                 alias: string_from_ptr(c_repo.alias),
                 user_name: string_from_ptr(c_repo.userName),
             };
-            res.push(r_repo);
+            repos_v.push(r_repo);
         }
         let repos_rawp = &mut repos;
         zypp_agama_sys::free_repository_list(
             repos_rawp as *mut _ as *mut zypp_agama_sys::RepositoryList,
         );
-    }
 
-    res
+        let res = if status.state == zypp_agama_sys::Status_STATE_STATE_SUCCEED {
+            Ok(repos_v)
```

Comment:
JFYI: yeah, as I kind of expect it will be more common, I planned to have that state to result helper also with parameter and additional parameter like `fn status_to_result_void<A>(mut status: zypp_agama_sys::Status, result: A) -> Result<A, crate::ZyppError>` to avoid having similar code at multiple places, but it can wait till we have more usage of such helper.

Reply:
Yes, Status will surely get more helpers along the way